server/events_controller.go

```diff
@@ -169,7 +169,7 @@ func (e *EventsController) handleBitbucketServerPost(w http.ResponseWriter, r *h
 		e.respond(w, logging.Error, http.StatusBadRequest, "Unable to read body: %s %s=%s", err, bitbucketServerRequestIDHeader, reqID)
 		return
 	}
-	if len(e.BitbucketWebhookSecret) > 0 {
```

Comment:
I'd like to pull this logic into a special check ahead of the rest of the code. I want to do this because I don't want there to be more conditionals where we're checking the webhook secret. I think that it might be confusing one day and we might refactor in a way that attackers could bypass the webhook secret checking.

```go
	if eventType == bitbucketserver.DiagnosticsPingHeader {
		// Specially handle the diagnostics:ping event because Bitbucket Server
		// doesn't send the signature with this event for some reason.
		e.respond(w, logging.Info, http.StatusOK, "Successfully received %s event %s=%s", eventType, bitbucketServerRequestIDHeader, reqID)
		return
	}
```